ejb3/src/main/java/org/jboss/as/ejb3/subsystem/deployment/AbstractEJBComponentRuntimeHandler.java

```diff
@@ -321,4 +348,136 @@ protected String resolveRuntimeName(final OperationContext context, final PathEl
+            addScheduleDetailString(schedNode, sched.getSecond(), SECOND.getName());
+            addScheduleDetailString(schedNode, sched.getTimezone(), TIMEZONE.getName());
+            addScheduleDetailDate(schedNode, sched.getStart(), START.getName());
+            addScheduleDetailDate(schedNode, sched.getEnd(), END.getName());
+        } catch (IllegalStateException e) {
+            // ignore
+        } catch (NoSuchObjectLocalException e) {
+            // ignore
+        } catch (EJBException e) {
+            logTimerFailure(componentName, e);
+        }
+    }
+
+    private static void addCalendarTimer(Timer timer, ModelNode timerNode, final String componentName) {
+        try {
+            final ModelNode detailNode = timerNode.get(CALENDAR_TIMER.getName());
+            boolean b = timer.isCalendarTimer();
+            detailNode.set(b);
+        } catch (IllegalStateException e) {
+            // ignore
+        } catch (NoSuchObjectLocalException e) {
+            // ignore
+        } catch (EJBException e) {
+            logTimerFailure(componentName, e);
+        }
+    }
+
+    private static void addPersistent(Timer timer, ModelNode timerNode, final String componentName) {
+        try {
+            final ModelNode detailNode = timerNode.get(PERSISTENT.getName());
+            boolean b = timer.isPersistent();
+            detailNode.set(b);
+        } catch (IllegalStateException e) {
+            // ignore
+        } catch (NoSuchObjectLocalException e) {
+            // ignore
+        } catch (EJBException e) {
+            logTimerFailure(componentName, e);
+        }
+    }
+
+    private static void addInfo(Timer timer, ModelNode timerNode, final String componentName) {
+        try {
+            final Serializable info = (timer instanceof TimerImpl) ? ((TimerImpl) timer).getCachedTimerInfo() : timer.getInfo();
+            if (info != null) {
+                final ModelNode detailNode = timerNode.get(INFO.getName());
+                detailNode.set(info.toString());
+            }
+        } catch (IllegalStateException e) {
+            // ignore
+        } catch (NoSuchObjectLocalException e) {
+            // ignore
+        } catch (EJBException e) {
+            logTimerFailure(componentName, e);
+        }
+    }
+
+    private static void addScheduleDetailString(ModelNode schedNode, String value, String detailName) {
+        final ModelNode node = schedNode.get(detailName);
+        if (value != null) {
+            node.set(value);
+        }
+    }
+
+    private static void addScheduleDetailDate(ModelNode schedNode, Date value, String detailName) {
+        final ModelNode node = schedNode.get(detailName);
+        if (value != null) {
+            node.set(value.getTime());
+        }
+    }
+
+    private static void logTimerFailure(final String componentName, final EJBException e) {
+        ROOT_LOGGER.failToReadTimerInformation(componentName);
+    }
```

Comment:
Maybe we can include the exception message in the log, if the stack trace is too long for a warning log.

Reply:
This method was moved here from the TimerAttributeDefinition class - but I agree with you - I see no reason to omit the exception.